[MFTF] MSI-1589: Fixing broken tests for MSI - Part 2

ivy00johns opened the pull request:
### Description
This PR contains the following improvements to stabilize existing Tests:
* Added a new Suite to isolate Tests dealing with the Sort Order and Pagination of the grid pages.
* Switched the "multi_mode" group tag with "sort_order_test".
* Replace the "skip" group with the `<skip></skip>` annotation.
* Added the "multi_mode" or "single_mode" group to tests that were missing them.
* Updated several tests that contained flaky action groups.

### Fixed Issues (if relevant)
1. Unskipped the following Tests:
* `AdminCreateOrderWithDownloadableProductDefaultStockTest.xml`
* `GuestCustomerOrderedGroupedProductOnSingleStockModeFromHomepageTest.xml`

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Updating skip group to use the correct skip annotation Tag.
- Skipping Tests due to know bugs in MSI.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Adding a conditionalClick to a Test to deal with a modal that needs to be dismissed.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Adding additional waitForPageLoad actions to address flaky behaviour.
- Adding a scrollToTopOfPage action to address a potential UI overlap issue.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Replacing skip group with the new <skip> annotation tag.
- Renaming the group "multi_source" with "multi_mode" to place it in the correct suite.
- Adding the group "multi_mode" to a few Tests to place them in the correct suite.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Adding a new suite for Tests that deal with the Grid Sorting and Pagination features.
- Adding group tag to the applicable Sort Order Tests.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Replacing "multi_mode" with "single_mode" in 1 test.
- Removing the "skip" tag from a test that I updated. The action group added a flaky step and there are multiple copies of the action group. Had to replace with actions instead.
- Updating a bad selector.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Removing "skip" tag from a Test I updated.
- Action Group was flaky, had to replace it with the steps.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Enabling inventory in the before steps.

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Enabling stock management in multiple tests.

ivy00johns commented:
Newest running build: http://10.234.148.146:8080/job/Functional-Tests-CE/1379/

ivy00johns commented:
Allure report for my latest build: http://10.234.148.146:8080/job/Functional-Tests-CE/1379/allure/
+ Failing: 3 non-MSI Tests
+ Broken: 1 MSI Test, cannot replicate locally

Investigating the broken tests.

ivy00johns committed:
Merge branch '2.3-develop' into 1589-Fixing-Broken-Tests-For-MSI

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: Skip tests failing because of magento-engcom/msi#1597

okolesnyk committed:
Merge branch '2.3-develop' of github.com:magento-engcom/msi into 1589-Fixing-Broken-Tests-For-MSI

okolesnyk commented:
Jenkins Functional Tests (MFTF): https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/4454/

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: Unskiping tests failing because of magento-engcom/msi#1597

okolesnyk commented:
https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/4463/

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: test fix

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: test fixes

okolesnyk committed:
Merge branch '2.3-develop' into 1589-Fixing-Broken-Tests-For-MSI

okolesnyk commented:
Latest build: https://m2build-ur.devops.magento.com/job/All-User-Requested-Tests/4533/

ivy00johns committed:
Merge remote-tracking branch 'remotes/origin/2.3-develop' into 1589-Fixing-Broken-Tests-For-MSI

ivy00johns committed:
magento-engcom/msi#1576-MSI-MFTF-Test-Cases-7

- Correcting a Checkout selector issue with the Test.
- Adding the correct stock reference to the Test.

ivy00johns committed:
Merge remote-tracking branch 'origin/1589-Fixing-Broken-Tests-For-MSI' into 1589-Fixing-Broken-Tests-For-MSI

okolesnyk committed:
Merge branch '2.3-develop' into 1589-Fixing-Broken-Tests-For-MSI

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: skip tests

okolesnyk committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI: update tests

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Correcting selectors.

ivy00johns committed:
Merge remote-tracking branch 'origin/1589-Fixing-Broken-Tests-For-MSI' into 1589-Fixing-Broken-Tests-For-MSI

ivy00johns committed:
magento-engcom/msi#1589-Fixing-Broken-Tests-For-MSI

- Increasing timeout for the checkout page.
- Replacing Stock that uses the Main Website.

ivy00johns commented:
Latest build: http://10.234.205.236:8080/job/Functional-Tests-EE/1165/
